erpnext/schools/doctype/student_applicant/student_applicant.js

```diff
@@ -42,6 +42,22 @@ frappe.ui.form.on("Student Applicant", {
 	}
 });
 
+frappe.ui.form.on('Student Guardian', {
```

Comment:
You can also do the same via `frm.add_fetch()` in the setup. For ex.
`frm.add_fetch("link_field_to_fetch_from", "field_in_linked_doctype", "field_to_map")`